### DISCUSSION TITLE
Members in a role?

### DISCUSSION THREAD

**Discussion by thetayloredman** [serenity-rs/serenity#2370] [Q&A] [answered]
Does Serenity expose a method of obtaining all of the members with a given role, or do I have to iterate over all members and see if they have a certain role?

**Comment by kangalio** [comment]
I think you need to do it manually

**Reply by thetayloredman** [reply]
With a loop over every member? Seems inefficient

**Reply by kangalio** [reply, marked as answer]
Discord has no built-in API endpoint to get members of a role, so a manual loop is the only way. Other libs do it the same way, e.g. [Discord.js](https://github.com/discordjs/discord.js/blob/8f6df90035e964d8779a6aab716c2f7f138975d5/src/structures/Role.js#L197), [Discord.py](https://github.com/Rapptz/discord.py/blob/ab287e71b8961cbe245e60ad220b7d0be9fd0b8a/discord/role.py#L373)

**Reply by thetayloredman** [reply]
> Discord has no built-in API endpoint to get members of a role, so a manual loop is the only way. Other libs do it the same way, e.g. [Discord.js](https://github.com/discordjs/discord.js/blob/8f6df90035e964d8779a6aab716c2f7f138975d5/src/structures/Role.js#L197), [Discord.py](https://github.com/Rapptz/discord.py/blob/ab287e71b8961cbe245e60ad220b7d0be9fd0b8a/discord/role.py#L373)

Considering I need to build a map of *every* role to *every* member, I guess I can do it that way. Thanks@
